src/dbnode/storage/database.go

```diff
@@ -474,7 +477,10 @@ func (d *db) AssignShardSet(shardSet sharding.ShardSet) {
 		//
 		// These small bootstraps can significantly delay topology changes as they prevent
 		// the nodes from marking themselves as bootstrapped and durable, for example.
-		d.queueBootstrapWithLock()
+		if !d.queueBootstrapWithLock() {
+			// enable file ops if bootstrap was not queued.
+			d.mediator.EnableFileOps()
```

Comment:
Can we be absolutely sure this will always get reenabled? Eg. could it be that 
`receivedNewShards = true` => we disable file ops above, and then
`d.bootstraps == 0` => we leave file ops disabled for good?

Reply:
that's why I've changed the return type for `queueBootstrapWithLock` so now it returns `bool` if bootstrap was queued  so we can `EnableFileOps` when bootstrapping was not queued.

Reply:
I'll try to phrase the question differently: in case when bootstrap is enqueued (i.e. `queueBootstrapWithLock()` returns `true`), will file ops get reenabled (and where)?

Reply:
yes, when bootstrap completes: https://github.com/m3db/m3/blob/3e78412b946dc82c320654ff477d8c4aa5a69e9c/src/dbnode/storage/bootstrap.go#L139

Reply:
Then I guess this would make sense, but this brings us to another issue - say a bootstrapped node is receiving a few shards from some node that is being removed, and this takes n hours. Would it mean that flushing gets disabled for those n hours? I think this would be a major issue.

Reply:
Yes, and this is an issue we already have, this change just adds additional safeguards so that we don't panic during these situations. I'm also working on a separate PR where different approach is taken to try getting rid of this issue.